dataRetriever/dataPool/proofsCache/proofsCache.go

```diff
@@ -13,22 +12,26 @@ type proofNonceMapping struct {
 }
 
 type proofsCache struct {
-	mutProofsCache sync.RWMutex
-	proofsByNonce  []*proofNonceMapping
-	proofsByHash   map[string]data.HeaderProofHandler
+	mutProofsByNonce     sync.RWMutex
+	proofsByNonceBuckets []*proofNonceBucket
```

Comment:
Alternatively, maybe `sync.Map` is useful?

Reply:
the way `proofsByNonceBuckets` is implemented right now, it needs order, so it doesn't fit very well